api/role.go

```diff
@@ -0,0 +1,44 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/ONSdigital/log.go/log"
+	"github.com/gorilla/mux"
+)
+
+//GetRoleHandler is a handler that gets a role by its ID from MongoDB
+func (api *API) GetRoleHandler(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	vars := mux.Vars(req)
+	roleID := vars["id"]
+	logdata := log.Data{"role-id": roleID}
+
+	//get role from mongoDB by id
+	role, err := api.permissionsStore.GetRole(ctx, roleID)
+	if err != nil {
```

Comment:
A `http.StatusNotFound` is returned regardless of the error. I think we should only be returning a 404 if the error is of type `errRoleNotFound`, otherwise return a `http.StatusInternalServerError`